documentation/testing_plans.md

```diff
@@ -0,0 +1,808 @@
+```
+
+Because the plan invokes the `get_targets` function, which accesses Bolt's
+inventory to retrieve `Target` objects, the tests must include an inventory. You
+can override the default inventory in `BoltSpec` using the `inventory_data`
+function.
+
+By default, `BoltSpec` tests run in `execute_any_plan` mode, which means the
+`patch::get_targets` plan is run in each test. For tests where it is not
+necessary to make assertions about the sub-plan, you can stub the sub-plan and
+have it return a set value.
+
+Save the following tests to `patch/spec/plans/init_spec.rb`:
+
+```ruby
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/plans'
+
+describe 'patch' do
+  include BoltSpec::Plans
+
+  # Inventory that is loaded by Bolt with targets that can be referenced
+  # by name in stubs and mocks
+  def inventory_data
+    {
+      'groups' => [
+        {
+          'name'    => 'agents',
+          'targets' => ['agent-target']
+        },
+        {
+          'name'    => 'agentless',
+          'targets' => ['agentless-target']
+        },
+        {
+          'name'    => 'empty',
+          'targets' => []
+        }
+      ]
+    }
+  end
+
+  # Configure Puppet and Bolt before running any tests
+  before(:all) do
+    BoltSpec::Plans.init
+  end
+
+  # Set the return value of puppet_agent::version for each target
+  # Stub the patch and puppet_agent::install tasks
+  before(:each) do
+    allow_task('puppet_agent::version').return_for_targets({
+      'agent-target'     => { 'version' => '7.0.0' },
+      'agentless-target' => { 'version' => nil }
+    })
+
+    allow_task('patch')
+    allow_task('puppet_agent::install')
+  end
+
+  it 'groups targets by puppet-agent status' do
+    result = run_plan('patch', 'targets' => 'all')
+
+    expect(result.value['agents'].map(&:name)).to eq(['agent-target'])
+    expect(result.value['agentless'].map(&:name)).to eq(['agentless-target'])
+  end
+
+  it 'returns an empty array if a group has no targets' do
+    result = run_plan('patch', 'targets' => 'empty')
+
+    expect(result.value['agents']).to eq([])
+    expect(result.value['agentless']).to eq([])
+  end
+
+  it 'runs the patch task on targets with agents' do
+    expect_task('patch').with_targets(['agent-target'])
+
+    run_plan('patch', 'targets' => 'all')
+  end
+
+  it 'does not run the patch task if there are no targets with agents' do
+    expect_task('patch').not_be_called
+
+    run_plan('patch', 'targets' => 'agentless')
+  end
+
+  it 'runs the puppet_agent::install task on agentless targets' do
+    expect_task('puppet_agent::install').with_targets(['agentless-target'])
+
+    run_plan('patch', 'targets' => 'all')
+  end
+
+  it 'does not run the puppet_agent::install task if there are no agentless targets' do
+    expect_task('puppet_agent::install').not_be_called
+
+    run_plan('patch', 'targets' => 'agents')
+  end
+end
+```
+
+Run the tests:
+
+```shell
+bundle exec rake spec
+```
```

Comment:
Could you add a related info section here? The backticks are throwing my suggestion off:

```
📖 Related information

- For more information on the available mocks and stubs, see the [BoltSpec reference](boltspec_reference.md).
```